### ISSUE TITLE
BUG-000124254 On Crowdsource Reporter web application and the service that used shared to everyone, Anonymous users are able to submit a report, while logged in as a viewer users are unable to submit a report

### ISSUE THREAD

**Issue by allisonmuise** [Esri/crowdsource-reporter#448]
<div><div><p>1. publish map service to ArcGIS server then share it to the portal for ArcGIS</p><p>2. create a group and share the published service to this group plus share it to everyone</p><p>3. use the shared map and the group to create Crowdsource Reporter</p><p>4. login into Crowdsource Reporter using Anonymous users you will be able to submit a report</p><p>5. login using level one user you will face an error that you don&#39;t have permissions to submit a report</p><p><br></p><p>This will be a very misunderstanding for the users as to how the Anonymous users able to submit a report without using any license but the level one users will not be able to submit a report.</p><p><br></p><p>Vm details,</p><p>Name : Supt0009485</p><p><br></p><p>Portal for ArcGIS username and password as admin : ( siteadmin for username and the password siteadmin)</p><p><br></p><p>Group name : ABC</p><p><br></p><p>Web-application name : ABC</p><p><br></p><p>Web application URL <a href="https://supt0009485.esri.com/portal/apps/CrowdsourceReporter/index.html?appid=793b2dfaa625474a993f024b7418ae98" target="_blank" style="color: rgb(0, 102, 204);"><u>https://supt0009485.esri.com/portal/apps/CrowdsourceReporter/index.html?appid=793b2dfaa625474a993f024b7418ae98</u></a></p><p><br></p><p>Level one username : test0001 and the password test00011</p><p><br></p><p>The confusion here is that users who are not logged in can submit the report, but user&#39;s logged in as level 1 viewer cannot.</p><p><br></p><p>Tested on portal for ArcGIS 10.7.1 and we faced same behavior</p></div><div>
<span><b>Salesforce ID:</b> <a href="https://qamonitor/sftriage?sfid=BUG-000124254">BUG-000124254</a></span><br/>
<span><b>Salesforce Submitter:</b> Qussai ALBattah</span><br/>
<span><b>Salesforce Submit Date:</b> 8/05/2019 12:56 PM</span><br/>
<span><b>Salesforce Bug Type:</b> Failure/Error</span><br/>
<span><b>Salesforce Severity:</b> Medium</span><br/>
<span><b>Repro Data:</b> (n/a)</span><br/>
<span><b>Work Around:</b> <p>login using higher role such as editor ​role</p></span><br/>
<span><b>Product:</b> Portal for ArcGIS</span><br/>
<span><b>Functional Category:</b> Portal Core</span><br/>
<span><b>Client Platform:</b> (n/a)</span><br/>
<span><b>Version Found:</b> 10.6.1</span><br/>
<span><b>Planned Version Fixed:</b> (n/a)</span><br/>
<span><b>Comment:</b> (n/a)</span><br/>
</div></div>

**Comment by allisonmuise** [comment]
3149

**Comment by ashishchoure** [comment]
@allisonmuise 

- **As per the current implementation**
    - The anonymous user and user's logged in through google plus and twitter can always submit the report.
    - Any user logged in through AGOL and having level greater than 1 with `"canEditFeatures" : true ` can submit the reports. And users with "**Level1**"  can never submit the report as he is viewer.

- So, as per the above comment since you were logged in as the "**Level1**" user, you were not able to submit the reports.

>The confusion here is that users who are not logged in can submit the report, but user's logged in as level 1 viewer cannot.

- **Possible solution to fix the above problem** 
   - We can add new configuration parameter "**allowEditsToAnonymousUser**" with the default value as **true**
  - This will not impact the existing users (Backward compatibility) and we can set the value to **false** to restrict the anonymous user's from submitting the reports.


Please let us know your thoughts.

**Comment by allisonmuise** [comment]
@ashishchoure This should not need to be a configuration parameter. A user who is signed in using an account they have paid for should never have lower permissions than an anonymous user. Could you look at how the other configurable app templates manage this scenario? I believe @MikeTschudi has handled this for Crowdsource Polling.

**Comment by ashishchoure** [comment]
@allisonmuise

I will see the behavior of Crowdsource Polling and update you at the earliest.

**Comment by MikeTschudi** [comment]
@ashishchoure, please look in Crowdsource Polling's template.js file, [`_checkSignIn()` function](https://github.com/Esri/crowdsource-polling/blob/438e383567d0824920ee8d3a94ab59226a974767/js/template.js#L267) and the [response handling](https://github.com/Esri/crowdsource-polling/blob/438e383567d0824920ee8d3a94ab59226a974767/js/template.js#L141).  It queries the app or user's access level for permission to use a private configuration of CP.  It's a different problem that this issue, but code may be useful.

**Comment by ashishchoure** [comment]
@MikeTschudi  

Thanks for sharing the details.
- The Crowd Source Reporter application is build on the same base as that of Crowd Source Polling. We found the same code in template.js file of reporter application.

@allisonmuise 
- We went through the code of Crowd Source Polling. The polling app is behaving same as reporter app. It allows Anonymous users to submit the comments and it restrict logged in users who don't have editing rights.
- The only difference is that we restrict "**Level 1**" user (as he is viewer) from performing edits and check editing rights for users with higher level. On the other hand polling application checks for editing rights for all the logged in user.

The editing functionality in crowdsource reporter is implemented as per the ticket [252 ](https://github.com/Esri/crowdsource-reporter/issues/252#issuecomment-263989211).

If required we can schedule a separate meeting to discuss this more as per your convenience.

**Comment by allisonmuise** [comment]
@ashishchoure We have confirmed that Polling has the same issue and I'll be adding an issue in that repo to fix it, too. I have confirmed that Basic Viewer handles the permissions correctly.
To test:
1. Create a map with an editable layer. 
2. Share the map and layer with the organization.
3. Use the map to configure a Basic Viewer app with the Editor tool enabled. Share the app with the organization.
4. Access the app using an account with editing capabilities. Note that the features can be edited. 
5. Access the app using an account without editing capabilities (Level 1, Viewer, or any custom level 2 account with editing capabilities disabled) and note that features can not be edited. 
6. Share the layer with Everyone and repeat steps 4 & 5. Note that both accounts can now edit the features because the layer can be edited by anyone.

The tests should be: Is the editable layer shared with Everyone? If yes, everyone should be able to edit. If no, only accounts with the 'edit' capability should be able to edit features.
